### PR TITLE
Client certificate provider

### PR DESCRIPTION
**Pull request by et1975**
For chocolatey/choco#399

### CODE REVIEW

**Comment by ferventcoder** [comment]
And now I'm watching this repository as well.

File: src/Core/Http/HttpClient.cs [hunk 1]
```diff
@@ -82,6 +83,19 @@ public static ICredentialProvider DefaultCredentialProvider
             }
         }
 
+        // TODO: Get rid of this. Horrid to have static properties like this especially in a code path that does not look thread safe.
```
Comment (ferventcoder) [inline src/Core/Http/HttpClient.cs:86]: ?

Comment (ferventcoder) [inline src/Core/Http/HttpClient.cs:86]: Copy pasta?

Comment (et1975) [inline src/Core/Http/HttpClient.cs:86]: I actually agree with the comment and left it on purpose, the static properties (and consequently entire IHttpClient) are bad. I actually tried to implement it outside of NuGet client at first, and again by passing the dependencies in, but various HttpClient bits violate LSP left and right and realized the static properties is the only way for this to work w/o rewriting everything that touches HttpClient.

Comment (ferventcoder) [inline src/Core/Http/HttpClient.cs:86]: I also agree that there are interesting things done in NuGet.

**Pull request merged**